specs/core/0_beacon-chain.md

```diff
@@ -1412,34 +1407,48 @@ For convenience, we provide the interface to the contract here:
 
 ## On genesis
 
-A valid block with slot `GENESIS_SLOT` (a "genesis block") has the following values. Other validity rules (e.g. requiring a signature) do not apply.
+When enough full deposits have been made to the deposit contract a `Eth2Genesis` log is emitted. The genesis block and state are defined by `get_genesis_beacon_block` and `get_genesis_beacon_state` where:
+
+* `genesis_validator_deposits` is the list of deposits, ordered chronologically, up to and including the deposit that triggered the `Eth2Genesis` log
+* `genesis_time` is timestamp specified in the `Eth2Genesis` log
+* `latest_eth1_data` is the `Eth1Data` when `Eth2Genesis` is emitted
 
 ```python
-{
-    slot=GENESIS_SLOT,
-    parent_root=ZERO_HASH,
-    state_root=GENESIS_STATE_ROOT,
-    randao_reveal=EMPTY_SIGNATURE,
-    eth1_data=Eth1Data(
-        deposit_root=ZERO_HASH,
-        block_hash=ZERO_HASH
-    ),
-    signature=EMPTY_SIGNATURE,
-    body=BeaconBlockBody(
+def get_genesis_beacon_block(state: BeaconState) -> BeaconBlock
```

Comment:
What `state` is passed in here? `get_genesis_beacon_state` requires the `genesis_block_root` so if this takes in the `genesis_beacon_state` then there's a circular dependency.

Reply:
Rewrote things so that it's crystal clear there is no circular dependency :) In short, `block_body` does *not* depend on `state`, and only `block_body` is referenced in `state`.